src/Npgsql/TypeHandlers/CompositeHandlers/CompositeParameterHandler.Generic.cs

```diff
@@ -0,0 +1,15 @@
+﻿using System.Reflection;
+using System.Threading.Tasks;
+using Npgsql.TypeHandling;
+
+namespace Npgsql.TypeHandlers.CompositeHandlers
+{
+    sealed class CompositeParameterHandler<T> : CompositeParameterHandler
+    {
+        public CompositeParameterHandler(NpgsqlTypeHandler handler, ParameterInfo parameterInfo)
+            : base(handler, parameterInfo) { }
+
+        public override async ValueTask<object?> Read(NpgsqlReadBuffer buffer, bool async)
```

Comment:
Unless I'm mistaken, this can complete synchronously - so maybe remove the `async` to avoid the overhead of the state machine?

Reply:
Makes sense, will do.

Reply:
Any reason not to do it? It seems pretty trivial... And the fact that we have boxing should make give up on other perf things...